python/ray/dashboard/subprocesses/module.py

```diff
@@ -47,12 +52,30 @@ class SubprocessModule(abc.ABC):
     def __init__(
         self,
         config: SubprocessModuleConfig,
+        parent_process_pid: int,
     ):
         """
         Initialize current module when DashboardHead loading modules.
         :param dashboard_head: The DashboardHead instance.
         """
         self._config = config
+        self._parent_process_pid = parent_process_pid
+        # Lazy init
+        self._gcs_aio_client = None
+        self._parent_process_death_detection_task = None
+
+    async def _detect_parent_process_death(self):
+        """
+        Detect parent process death by checking if ppid is still the same.
+        """
+        while True:
+            ppid = os.getppid()
```

Comment:
```
Return the parent’s process id. When the parent process has exited, on Unix the id returned is the one of the init process (1), on Windows it is still the same id, which may be already reused by another process.
```

This approach won't work on Windows.

Try `multiprocessing.parent_process().is_alive()` or pipe.

Reply:
Use `psutil` instead.